src/azure-cli/azure/cli/command_modules/maps/custom.py

```diff
@@ -96,13 +100,17 @@ def maps_account_update(client,
     if disable_local_auth is None:
         disable_local_auth = False
     maps_account_update_parameters = {}
-    maps_account_update_parameters['tags'] = tags
+    if tags is not None:
+        maps_account_update_parameters['tags'] = tags
     maps_account_update_parameters['kind'] = "Gen1" if kind is None else kind
     maps_account_update_parameters['disable_local_auth'] = False if disable_local_auth is None else disable_local_auth
-    maps_account_update_parameters['linked_resources'] = linked_resources
+    if linked_resources is not None:
+        maps_account_update_parameters['linked_resources'] = linked_resources
     maps_account_update_parameters['identity'] = {}
-    maps_account_update_parameters['identity']['type'] = type_
-    maps_account_update_parameters['identity']['user_assigned_identities'] = user_assigned_identities
+    if type_ is not None:
+        maps_account_update_parameters['identity']['type'] = type_
+    if user_assigned_identities is not None:
+        maps_account_update_parameters['identity']['user_assigned_identities'] = user_assigned_identities
     maps_account_update_parameters['sku'] = {}
     maps_account_update_parameters['sku']['name'] = name
```

Comment:
If `identity`, `sku`, `` has no attributes. It's value should be None.  If `kind` is None, you should not assign default value for it.